### DISCUSSION TITLE
read_only_fields => readonly_fields

### DISCUSSION THREAD

**Discussion by mmichealjroberts** [encode/django-rest-framework#7741] [Ideas & Suggestions]
Suggested Minor Improvement:

Inside serializers, change `read_only_fields` to `readonly_fields` to match the Django standard (e.g., when creating `admin.py` code).

**Comment by rpkilby** [comment]
On the one hand, consistency with Django is a +1, on the other, serializer fields have a `read_only` argument, and `read_only_fields` is more internally consistent.

That said, I'm indifferent to the change, and if we did want to do this, we could copy [`RenameAttributesBase`](https://github.com/carltongibson/django-filter/blob/master/django_filters/utils.py#L30-L75) from django-filter (which is based on Django's [`RenameMethodsBase`](https://github.com/django/django/blob/9a2a12d415e6d224797b0779c5af03deb8c45b0a/django/utils/deprecation.py#L30-L70)).

Speaking of, @carltongibson do you think it would be worth adding `RenameAttributesBase` to Django to complement `RenameMethodsBase`? It might be a useful addition, but I've always been concerned with the `__getattr__`/`__setattr__` overrides.

**Comment by lovelydinosaur** [comment]
I'm in favour of the internal consistency here.
